### PR TITLE
Event detection using bayer

### PR DESCRIPTION
**Pull request by fabito**

### CODE REVIEW

File: examples/imagr_event_detection/classify_images.cc [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  auto* input_tensor = interpreter->input_tensor(0);
+
+  CameraFrameFormat fmt{CameraFormat::kRaw,
+                        CameraFilterMethod::kBilinear,
+                        CameraRotation::k270,
+                        CameraTask::kWidth,
+                        CameraTask::kHeight,
+                        false,
+                        image->data()};
+
```
Comment (fabito) [inline examples/imagr_event_detection/classify_images.cc:128]: This is preprocessing logic:

- get frame from camera as bayer (kRaw) 324x324
- crop bayer image (125x23)
- convert bayer crop to grayscale
- resize grayscaled crop to event detection model input size (56x56)
- perform inference

Comment (fabito) [reply examples/imagr_event_detection/classify_images.cc:128]: @konstantinos-imagr

Comment (konstantinos-imagr) [reply examples/imagr_event_detection/classify_images.cc:154]: what prob we need to do:

1. get frame from camera as bayer (kRaw) 324x324
2. crop bayer image (125x23)
3. -
4. resize crop to event detection model input size (56x56)
5. reshape to 56x56x1
6. perform inference

File: examples/imagr_event_detection/isp_processing.hpp [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+
+uint8_t saturate_cast(float value) {
+   uint8_t maxValue = std::numeric_limits<uint8_t>::max();
+    if (value > maxValue)
+        return maxValue;
+    else
+        return static_cast<uint8_t>(value);
+}
+
+void ResizeNearestNeighbor(const uint8_t* src, int src_w, int src_h,
```
Comment (fabito) [inline examples/imagr_event_detection/isp_processing.hpp:18]: Most of the functions here were duplicated from `camera.cc`.

File: examples/imagr_event_detection/isp_processing.hpp
Comment (fabito) [inline examples/imagr_event_detection/isp_processing.hpp:1]: Most of the functions here were duplicated from camera.cc.

File: examples/imagr_event_detection/classify_images.cc (hunk 1 above)
Comment (fabito) [reply examples/imagr_event_detection/classify_images.cc:153]: I removed the grayscale conversion now

Comment (nigel-ga) [reply examples/imagr_event_detection/classify_images.cc:153]: just  for reference, the supplied BayerToGrayscale function will destroy the bayer pattern as it applies a coefficient to each of the bayer channels to balance them out.

Comment (Wal8800) [reply examples/imagr_event_detection/classify_images.cc:153]: @fabito Update: Dinos is trying to move this step `reshape to 56x56x1` as part of the model. Currently, he trained a model with that operation and now looking to convert the model into a tflite model. 

Curious, after removing the grayscale conversion, does the inference still work? Does the existing pipeline expand the dims of 56 x 56 to 56 x 56 x 1 already :thinking:

Comment (Wal8800) [reply examples/imagr_event_detection/classify_images.cc:160]: Ran the current changes locally, it seem like the inference runs succesfully without doing the extra `reshape to 56x56x1`, maybe it's not needed at all? The input tensor seem to take in a stream of bytes :thinking:  cc @konstantinos-imagr . 

The web client always show a grey image and 0.5 for something and nothing. I wonder what's causing this.

Comment (Wal8800) [reply examples/imagr_event_detection/classify_images.cc:160]: Corrected the colour on the web client's `script.js` by changing 

```
cv.cvtColor(mat, debayeredMat, cv.COLOR_BayerBG2BGRA);
```

to

```
cv.cvtColor(mat, debayeredMat, cv.COLOR_BayerRG2RGBA);
```

**Pull request closed without merging**